test/JDBC/expected/BABEL-5186-vu-verify.out

```diff
@@ -0,0 +1,2460 @@
+~~END~~
+
+
+-- Error: relation \"%s\" does not exist
+-- Simple batch with try catch
+BEGIN TRY
+    SELECT * FROM non_existent_table;
+END TRY
+BEGIN CATCH
+		SELECT 'Severity_' + cast(ERROR_SEVERITY() as nvarchar(500))
+					+ ' Error State_'+ cast(ERROR_STATE() as nvarchar(500))
+					+ ' Xact State_'+ cast(XACT_STATE() as nvarchar(500))
+					+ ' Error number_'+ cast(ERROR_NUMBER() as nvarchar(500))
+					+ ' Error Line number_'+ cast(ERROR_LINE() as nvarchar(500))
+					+ ' Error message_'+ cast(ERROR_MESSAGE() as nvarchar(500));
+END CATCH
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: relation "non_existent_table" does not exist)~~
+
+
+BEGIN TRY
+    EXEC('SELECT * FROM non_existent_table;');
+END TRY
+BEGIN CATCH
+		SELECT 'Severity_' + cast(ERROR_SEVERITY() as nvarchar(500))
+					+ ' Error State_'+ cast(ERROR_STATE() as nvarchar(500))
+					+ ' Xact State_'+ cast(XACT_STATE() as nvarchar(500))
+					+ ' Error number_'+ cast(ERROR_NUMBER() as nvarchar(500))
+					+ ' Error Line number_'+ cast(ERROR_LINE() as nvarchar(500))
+					+ ' Error message_'+ cast(ERROR_MESSAGE() as nvarchar(500));
+END CATCH
+GO
+~~START~~
+nvarchar
+Severity_16 Error State_1 Xact State_0 Error number_33557097 Error Line number_1 Error message_relation "non_existent_table" does not exist
+~~END~~
+
+
+-- Error: relation \"%s\" does not exist
+-- Simple batch with transaction inside try-catch
+BEGIN TRY
+    BEGIN TRAN
+        SELECT * FROM non_existent_table;
+END TRY
+BEGIN CATCH
+		SELECT 'Severity_' + cast(ERROR_SEVERITY() as nvarchar(500))
+					+ ' Error State_'+ cast(ERROR_STATE() as nvarchar(500))
+					+ ' Xact State_'+ cast(XACT_STATE() as nvarchar(500))
+					+ ' Error number_'+ cast(ERROR_NUMBER() as nvarchar(500))
+					+ ' Error Line number_'+ cast(ERROR_LINE() as nvarchar(500))
+					+ ' Error message_'+ cast(ERROR_MESSAGE() as nvarchar(500));
+END CATCH;
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: relation "non_existent_table" does not exist)~~
+
+
+SELECT @@trancount
+GO
+~~START~~
+int
+0
+~~END~~
+
+
+IF @@trancount > 0 ROLLBACK TRAN;
+GO
+
+BEGIN TRY
+    BEGIN TRAN
+        EXEC('SELECT * FROM non_existent_table;');
+END TRY
+BEGIN CATCH
+		SELECT 'Severity_' + cast(ERROR_SEVERITY() as nvarchar(500))
+					+ ' Error State_'+ cast(ERROR_STATE() as nvarchar(500))
+					+ ' Xact State_'+ cast(XACT_STATE() as nvarchar(500))
+					+ ' Error number_'+ cast(ERROR_NUMBER() as nvarchar(500))
+					+ ' Error Line number_'+ cast(ERROR_LINE() as nvarchar(500))
+					+ ' Error message_'+ cast(ERROR_MESSAGE() as nvarchar(500));
+END CATCH;
+GO
+~~START~~
+nvarchar
+Severity_16 Error State_1 Xact State_-1 Error number_33557097 Error Line number_1 Error message_relation "non_existent_table" does not exist
+~~END~~
+
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Uncommittable transaction is detected at the end of the batch. The transaction is rolled back.)~~
```

Comment:
I think we should not abort the transaction for this error if it is going to get caught in the catch block.
Can we check the transaction behaviour for this error in SQL Server for both cases, when it gets caught in catch block and when it does not.

Reply:
Discussed offline, this is an expected behaviour as per the implementation.